src/service.rs

```diff
@@ -18,13 +20,24 @@ use crate::{
     model::{Instance, InstanceStage},
 };
 
+/// Returns true if and only if the name is a valid instance name.
+///
+/// Instance name will be used as kubernetes's resource names, such as pod names, label names,
+/// hostnames and so on. So the same naming constraints should be applied to the instance name.
+/// See: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names.
+fn verify_instance_name(name: &str) -> bool {
+    static RE: Lazy<Regex> =
```

Comment:
Maybe it could be put outside? 😆  

We have the test!

Reply:
make sense